flows/definition/dependencies.go

```diff
@@ -59,10 +60,11 @@ func (d *dependencies) refresh(sa flows.SessionAssets, missing assets.MissingCal
 	for i, ref := range d.Flows {
 		a, err := sa.Flows().Get(ref.UUID)
 		if err != nil {
-			missing(ref)
-		} else {
-			d.Flows[i] = a.Reference()
+			// flows are the one thing that aren't allowed to be missing as we wouldn't know how to
+			// route from a subflow split with a missing flow
+			return errors.Wrapf(err, "unable to read %s", ref)
```

Comment:
this means we won't start a flow A which contains a subflow B if
* B doesn't exist
* B is unreadable (missing non-flow assets within flow B still ok)

@nicpottier

Reply:
Having references to flows that don't exist is a thing right now that isn't super uncommon. I think we currently run these just fail when we reach that node. I think we should probably still do the same or we are going to get into trouble.

Reply:
Just checked the old engine and it's behavior in this scenario looks a bit undefined, but also it's looking up the subflow without filtering by `is_archived` or `is_active` so it was way less likely to run into this scenario. The new engine currently lets the flow pass validation when running (it's treated like any other missing asset) - but then hard errors (i.e. `Resume` returns `error`) if a `EnterFlow` action tries to load it and can't.

Agree best to be forgiving for now so I think we can have a missing subflow result in a fatal error event.

Reply:
Updated so that flows treated like any other missing asset at inspection time - and at run time we generate a fatal error if we can't get the subflow.